compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -905,17 +907,61 @@ public void visit(BLangArrayType arrayTypeNode) {
             BArrayType arrType;
             if (arrayTypeNode.sizes.length == 0) {
                 arrType = new BArrayType(resultType, arrayTypeSymbol);
+                arrayTypeSymbol.type = arrType;
+                markParameterizedType(arrType, arrType.eType);
+                resultType = arrayTypeSymbol.type;
             } else {
-                int size = arrayTypeNode.sizes[i];
-                arrType = (size == UNSEALED_ARRAY_INDICATOR) ?
-                        new BArrayType(resultType, arrayTypeSymbol, size, BArrayState.UNSEALED) :
-                        (size == OPEN_SEALED_ARRAY_INDICATOR) ?
-                                new BArrayType(resultType, arrayTypeSymbol, size, BArrayState.OPEN_SEALED) :
-                                new BArrayType(resultType, arrayTypeSymbol, size, BArrayState.CLOSED_SEALED);
-            }
-            resultType = arrayTypeSymbol.type = arrType;
+                BLangExpression size = arrayTypeNode.sizes[i];
+                if (size instanceof BLangLiteral) {
```

Comment:
Is there a specific reason for this? Because if I don't specifically check for the BLangLiteral type in the if statement, my current logic won't stand as the else block assumes that it is receiving BLangLiteral Type.